git/git_test.go

```diff
@@ -1,13 +1,14 @@
 // Copyright 2023 Terramate GmbH
 // SPDX-License-Identifier: MPL-2.0
 
-//go:build linux
+//go:build linux || darwin
```

Comment:
I'm surprised this works :-)
Our tests generates a repository with a HEAD commit SHA that's known beforehand and this used to be an issue to work in non-linux envs.

Reply:
Was not intentional, I had to change this locally to debug the tests and forgot to change it back... But we can leave it in I suppose if it now works on the CI? On my Mac all tests in this file work as well.

Reply:
yes we can test on OSX as well, cool.